user-service/domains/repositories/user_repository.py

```diff
@@ -96,5 +96,36 @@ def add_following(self, user_id, following=[]):
         self.session.merge(user)
         self.session.commit()
         return user
+    """
+    Helper for updating a User object
 
+    :param username
+    :return: 
+    """
+    @check_id_exists(User, ["user_id", "username"])
+    @check_unique(User, User.username, ["username"])
+    def _update_user(self, user: User, username, streaming_status):
+        user.username = username
+        if (streaming_status):
+            user.streaming_status = streaming_status
+        return user
 
+    """
+    Updates a User object
+
+    :optional param username: str of username
+    :optional param user_id: str of uuid (must be able to be parsed to uuidv4)
+    :optional param streaming_status: str of streaming status
+    :return: User of updated user
+    """
+    @check_id_exists(User, ["user_id"])
+    def update_user(self, username=None, user_id=None, streaming_status=None):
+        user = self.session.get(User, user_id)
+        if (not username):
+            return user
+        # updated_user = User(username=username, user_id=user_id, streaming_status=streaming_status)
```

Comment:
just use this instead and check if fields are none default to their last field i.e.
username = username if username is not None else user.username

Reply:
i believe you wanted me to not do any patches and no errors if username doesn't exist

Reply:
yeah thats why you are setting username to either user.username (if there is no updated value) or username (if there is an updated value)

Reply:
sorry, i meant you told me to not do ANYTHING if username doesn't exist. So if there is no username and there is some streaming_status, then should I send a patch?